packages/lesswrong/server/search/elastic/ElasticSearchExporter.ts

```diff
@@ -0,0 +1,308 @@
+import { OnDropDocument } from "@elastic/elasticsearch/lib/helpers";
+import { htmlToText } from "html-to-text";
+import ElasticSearchClient from "./ElasticSearchClient";
+import { collectionNameToConfig, Mappings } from "./ElasticSearchConfig";
+import {
+  AlgoliaIndexCollectionName,
+  algoliaIndexedCollectionNames,
+} from "../../../lib/search/algoliaUtil";
+import {
+  AlgoliaIndexedCollection,
+  AlgoliaIndexedDbObject,
+} from "../utils";
+import {
+  CommentsRepo,
+  PostsRepo,
+  SequencesRepo,
+  TagsRepo,
+  UsersRepo,
+} from "../../repos";
+import { getCollection } from "../../../lib/vulcan-lib/getCollection";
+import Globals from "../../../lib/vulcan-lib/config";
+
+const HTML_FIELDS = [
+  "body",
+  "bio",
+  "howOthersCanHelpMe",
+  "howICanHelpOthers",
+  "plaintextDescription",
+  "description",
+];
+
+class ElasticSearchExporter {
+  constructor(
+    private client = new ElasticSearchClient(),
+  ) {}
+
+  async configureIndexes() {
+    for (const collectionName of algoliaIndexedCollectionNames) {
+      await this.configureIndex(collectionName);
+    }
+  }
+
+  async exportAll() {
+    await Promise.all(algoliaIndexedCollectionNames.map(async (collectionName) => {
+      await this.exportCollection(collectionName);
```

Comment:
nit: can this just be `await Promise.all(algoliaIndexedCollectionNames.map(this.exportCollection.bind(this))`?  (Or `.map((collectionName) => this.exportCollection(collectionName))`, for readability.)

I think the extra internal `await` might add something to the stack trace, but not sure if that's the only (intended) effect.